### ISSUE TITLE
[java] Consider splitting rule "NonThreadSafeSingleton"

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#7036]
Consider to split the rule [NonThreadSafeSingleton](https://docs.pmd-code.org/latest/pmd_rules_java_multithreading.html#nonthreadsafesingleton) into two separate rules:
* one rule to check for singletons
* another rule to check lazy initialization patterns

The `checkNonStaticFields` option decides whether we want to check singleton or lazy loading pattern in general. The rule name only covers the former, but otherwise I think the property distinguishes two valid usecases.

For `checkNonStaticMethods` the following two samples are equivalent:

```java

class Invoice {

  static Databse db;

  public short money;

  public void save() {
    if (db == null) {db = Database.connect("cloud");}
    db.saveInvoice(money);
  }
}
```

and

```java

class Invoice {

  static Databse db;

  public short money;

  private static void initDB() {
   if (db == null) {db= Database.connect("cloud");}
  }

  public void save() {
    initDB();
    db.saveInvoice(money);
  }
}
```
Whether the "singleton" classification is applicable to these two samples is debatable, but it fits both equally well.
I don't see a reason why would anyone want a violation reported for the second sample and not the first, so ignoring the property would be IMHO an improvement.

_Originally posted by @zbynek in https://github.com/pmd/pmd/issues/6864#issuecomment-4996965344_
